packages/schemas/src/schemaComponents.ts

```diff
@@ -19,7 +19,7 @@ export const componentsBySchema: Record<CoreSchemaType, InputComponent[]> = {
 	[CoreSchemaType.MemberId]: [InputComponent.memberSelect],
 	[CoreSchemaType.Vector3]: [InputComponent.confidenceInterval],
 	[CoreSchemaType.Null]: [],
-} as const;
+} as const satisfies Record<CoreSchemaType, InputComponent[]>;
```

Comment:
I changed this because I experimented with finding the `CoreSchemaType` by the `InputComponent`.
I realized this wasn't worth it in the end, so I didn't do anything with it, but this type is more `correct` so I say we keep it.

Reply:
for my own understanding, what makes this type more `correct`?

Reply:
I think (though @tefkah should validate this) that using `satisfies` rather than a type assertion here will still make sure that this is a `Record<CoreSchemaType, InputComponent[]>`, but also allow the compiler to keep the more specific information about what each property maps to. Previously the typescript compiler would decide the expression `componentsBySchema[CoreSchemaType.Vector3]` is `InputComponent[]`, but with this change it will infer the type `[InputComponent.confidenceInterval]`

Reply:
thank you both, this makes a lot of sense! 🧠